elasticsearch/sandboxes/heka/output/elasticsearch_bulk_api.lua

```diff
@@ -80,6 +83,18 @@ local req_headers = {
     ["connection"]      = "keep-alive",
 }
 
+local function basic_auth_string(params)
+    assert(type(params.username) == 'string')
```

Comment:
I appreciate the check but the error message it outputs isn't too helpful (without looking back at the code).  Also it doesn't check if params is a table so it could blow up before the assert.

Reply:
Just slowly getting used to the style. All valid points I can agree with.